spec/files/README.md

```diff
@@ -42,3 +42,39 @@ when `patchelf_compatible: true`, and by default fails with `NotImplementedError
 
 ### 2.3) Todo
 Update `why was this added?` after implementing `new_load_method`(also remove this and the last sentence;)
+
+
+## 3. libbuildid.so
+
+### 3.1) Source
+It is generated by the included Makefile.
+
+### 3.2) Why was this added?
+This is a library with the `.gnu.note.build-id` section. It is used to test note sections are correctly relocated along with note segments.
+
+
+## 4. contiguous-note-sections.elf
+
+### 4.1) Source
+It is generated by the included Makefile, using the included contiguous-note-section.ld and contiguous-note-section.s files. The original source can be found at https://github.com/NixOS/patchelf/tree/master/tests.
+
+### 4.2) Why was this added?
+This covers an edge case of when there are multiple note sections back-to-back with varying alignments. This test ensures that this does not cause the patcher to error and think there are gaps.
+
+
+## 5. empty-note
+
+### 5.1) Source
+This ELF is hand crafted to contain a PT_NOTE segment that has a non-zero size but does not have any sections within that segment.
+
+### 5.2) Why was this added?
+This tests backwards compatibility with older versions of the patchelf-compatible saver that could produce such binaries. This test ensures that this does not cause the patcher to error.
+
+
+## 6. libbig-dynstr.debug
+
+### 6.1) Source
+This was built from sources at https://github.com/NixOS/patchelf/tree/master/tests.
```

Comment:
can use a permalink (in case the files on NixOS/patchelf change someday)

Also add the commands for generating the binary (I guess it's just something like `autoconf` + `automake`, but good to document them so it's easier to be updated)

Reply:
Also added a "todo" to try simplify this executable down to something we can port to our Makefile. Upstream probably do a bit more than necessary (they generate an executable with 2000 functions), and the issue doesn't trigger under certain compiler setups.